packages/core/src/app/ovm/deciders/or-decider.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  Decider,
+  Decision,
+  ImplicationProofItem,
+  Property,
+} from '../../../types/ovm'
+import { CannotDecideError } from './utils'
+
+export interface OrDeciderInput {
+  properties: Property[]
+  witnesses: any[]
+}
+
+/**
+ * Decider that decides true if any of the provided properties evaluate to true.
+ */
+export class OrDecider implements Decider {
+  private static _instance: OrDecider
+
+  public static instance(): OrDecider {
+    if (!OrDecider._instance) {
+      OrDecider._instance = new OrDecider()
+    }
+    return OrDecider._instance
+  }
+
+  public async decide(
+    input: OrDeciderInput,
+    witness?: undefined,
+    noCache?: boolean
+  ): Promise<Decision> {
+    const decisions: Decision[] = await Promise.all(
+      input.properties.map((property: Property, index: number) => {
+        return this.decideWithoutThrowingCannotDecide(
+          property,
+          input.witnesses[index],
+          noCache
+        )
+      })
+    )
+
+    let trueDecision: Decision
+    let cannotDecide: boolean = false
+    const falseJustifications: ImplicationProofItem[] = []
+    for (const decision of decisions) {
+      if (!decision) {
+        cannotDecide = true
+        continue
+      }
+      if (decision.outcome) {
+        trueDecision = decision
+        break
+      } else {
+        falseJustifications.push(...decision.justification)
+      }
+    }
+
+    if (trueDecision) {
+      return this.getDecision(input, trueDecision)
+    }
+
+    if (cannotDecide) {
+      throw new CannotDecideError(
+        'At least one of the OR deciders could not decide and none returned true, so this cannot be decided.'
+      )
+    }
+
+    return this.getDecision(input, {
+      outcome: false,
+      justification: falseJustifications,
+    })
+  }
```

Comment:
This decider looks good to me! The only comment would be that most of the time I imagine we will often end up using NOT NOT AND in place of OR, but that doesn’t change the fact that this looks solid

Reply:
Yeah, I just made the OR to reduce nesting and [originally] I needed to support 3 sub-deciders, and AND only supports 2, so I made OR and made it support any number of inputs.